sql/src/main/java/io/crate/expression/operator/LikeOperator.java

```diff
@@ -60,33 +67,48 @@ public FunctionInfo info() {
             if (value == null) {
                 return this;
             }
-            return new CompiledLike(info, (String) value);
+            boolean ignoreCase = ((Literal<Boolean>) arguments.get(2)).value();
```

Comment:
Maybe add an assertion here that the second argument must be a literal due to us setting it implicitly based on `LIKE` / `ILIKE` instead of it being an argument supplied by a user.

(Makes me also wonder a bit, if having a separate ilike function would have been cleaner)